src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,165 @@
 package core.basesyntax;
 
-public class ArrayList<T> implements List<T> {
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
+public abstract class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        this.elements = new Object[DEFAULT_CAPACITY];
+        this.size = 0;
+    }
+
+    private void ensureCapacity() {
+        if (size == elements.length) {
+            int newCapacity = (int) (elements.length * 1.5 + 1);
+            Object[] newArray = new Object[newCapacity];
+            System.arraycopy(elements, 0, newArray, 0, elements.length);
+            elements = newArray;
+        }
+    }
+
     @Override
-    public void add(T value) {
+    public boolean add(T element) {
+        ensureCapacity();
+        elements[size++] = element;
+        return true;
+    }
 
+    @Override
+    public T get(int index) {
```

Comment:
Don't complicate if-else construction. The index check is repeated in `get`, `remove(int index)`, and `set` methods. Refactor it into a separate method to avoid code duplication.